### PR TITLE
Moved token generation to doctrine event listner thing

### PR DESCRIPTION
**Pull request by JeroenDeDauw**

### CODE REVIEW

File: src/UseCases/AddDonation/AddDonationUseCase.php [hunk 1]
```diff
@@ -76,52 +71,16 @@ public function addDonation( AddDonationRequest $donationRequest ): AddDonationR
...
-		$updateToken = $this->tokenGenerator->generateToken();
-
-		$this->authorizationUpdater->allowModificationViaToken(
-			$donationId,
-			$updateToken,
-			$this->tokenGenerator->generateTokenExpiry()
+		return AddDonationResponse::newSuccessResponse(
+			$donation,
+			$tokens->getUpdateToken(),
+			$tokens->getAccessToken()
```
Comment (KaiNissen) [inline src/UseCases/AddDonation/AddDonationUseCase.php]: We could pass the instance of `DonationTokens` instead of two strings here.

**Comment by JeroenDeDauw** [comment]
Meh. So now the CI breaks whenever we do a composer update apparently... @gbirke how did you fix this when it happened last time? @KaiNissen just ignore for this PR please. Tests run locally, and there is no indication on TravisCI anything from this PR is wrong either.

File: src/UseCases/AddDonation/AddDonationUseCase.php (hunk 1 above)
Comment (JeroenDeDauw) [inline src/UseCases/AddDonation/AddDonationUseCase.php]: Yeah. I consciously did not include that in this PR though.

**Comment by KaiNissen** [comment]
Yeah, already tried the tests locally.

**Pull request merged**